### ISSUE TITLE
ruler: Use `notifier` Prometheus package when decoupled from config package.

### ISSUE THREAD

**Issue by bwplotka** [thanos-io/thanos#660]
Rationales: Ensure similar logic and usage of powerful relabelling in Thanos Ruler.

Acceptance Criteria
* We don't mainain `pkg/alert` package and instead use `notifier` from Prometheus codebase.
* We don't depend on `config` package that depends on lots of 3rd party packages (k8s, consul, ec2, etc)

Right now our code mimics notifier.

**Comment by markmsmith** [comment]
Is this work still planned, or is full relabeling in Ruler already supported in some other way?

**Comment by FUSAKLA** [comment]
Hi, I believe for now there are the `--alert.label-drop=` and `--label=` flags which I'd say cover most of the use cases?
Let us knou if you are blocked by this and those labels does not suffice you and we can look more into it.
Right now I'm not aware of anyone working on this.

**Comment by markmsmith** [comment]
We're still in the early stages of figuring out our Thanos alerting setup, so I'm not sure yet if we'll need richer relabeling or not.  I was just curious since it seemed like the PR linked above was pretty much there, it just needed a final 👍.

**Comment by lilic** [comment]
@bwplotka  Can we remove this note that links to issue in the docs? https://github.com/thanos-io/thanos/blob/master/docs/components/rule.md#ruler-ha If not all is implemented do you think it makes sense to open a new issue about this?

**Comment by bwplotka** [comment]
We might need another issue if we need such, so far we did not seen the need ):  Anyway the comment in doc is confusing (:

**Comment by sjentzsch** [comment]
We require full alert relabeling within Thanos Rule.
To move forward, we would consider a contribution, but would need some assistance/guidance on how to move forward.
Maybe @bwplotka , how would you proceed or what would you suggest as next step?

**Comment by yeya24** [comment]
IIUC, we still want to use `notifier` package rather than maintaining our own `alert` package? I will reopen this issue, feel free to comment and close it if it is unnecessary anymore
